activerecord/test/cases/encryption/contexts_test.rb

```diff
@@ -11,24 +11,26 @@ class ActiveRecord::Encryption::ContextsTest < ActiveRecord::EncryptionTestCase
     ActiveRecord::Encryption.config.support_unencrypted_data = true
 
     @post = EncryptedPost.create!(title: "Some encrypted post title", body: "Some body")
-    @clean_title = @post.title
+    @title_cleartext = @post.title
+    @title_ciphertext = @post.ciphertext_for(:title)
   end
 
   test ".with_encryption_context lets you override properties" do
     ActiveRecord::Encryption.with_encryption_context(encryptor: ActiveRecord::Encryption::NullEncryptor.new) do
-      assert_protected_encrypted_attribute(@post, :title, @clean_title)
+      assert_equal @title_ciphertext, @post.reload.title
+
       @post.update!(title: "Some new title")
     end
 
-    assert_equal "Some new title", @post.title
+    assert_equal "Some new title", @post.title_before_type_cast
```

Comment:
Why do we need to do `title_before_type_cast` here? Shouldn't just `#title` work?

Reply:
`#title` still works here too.  But `#title` would also work if `with_encryption_context` were removed (or broken), because `#title` would decrypt the ciphertext as normal.  Checking `title_before_type_cast` is a step closer to verifying that the title was encrypted with `NullEncryptor` (and that `with_encryption_context` worked).

I could also use `assert_not_encrypted_attribute` here, but I feel like that doesn't fit semantically with `with_encryption_context`.